pkg/job_controller/service.go

```diff
@@ -162,11 +162,28 @@ func (jc *JobController) FilterServicesForReplicaType(services []*v1.Service, re
 	return result, nil
 }
 
+// calculateServiceSliceSize compare max pod index with desired replicas and return larger size
+func calculateServiceSliceSize(services []*v1.Service, replicas int) int {
```

Comment:
is it necessary? in `GetServiceSlices` function, serviceSlices will be re-allocated when index> size

Reply:
If calculateServiceSliceSize is not used, there will exist an error "index out of range" when scaling in workers and services (ie., update worker replicas from 2 to 1).